code/modules/polling/poll_editor_panel.dm

```diff
@@ -0,0 +1,96 @@
+/datum/poll_editor_panel
+	var/last_error = null
+
+/datum/poll_editor_panel/New(mob/user)
+	..()
+	// TODO: pass in current poll to edit it
+
+/datum/poll_editor_panel/ui_state(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/poll_editor_panel/ui_status(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/poll_editor_panel/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "PollEditorPanel")
+	ui.open()
+
+/datum/poll_editor_panel/ui_static_data(mob/user)
+	. = list(
+		"serverOptions" = list("Local", "Global", "RP Only")
+	)
+
+/datum/poll_editor_panel/ui_data(mob/user)
+	. = list(
+		"lastError" = src.last_error
+	)
+
+/datum/poll_editor_panel/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if (.)
+		return
+
+	switch (action)
+		if ("save")
+			USR_ADMIN_ONLY
+
+			var/list/poll
+
+			try
+				// TODO: html_encode?
```

Comment:
Mostly in a draft state because I want to investigate this. It was in the earlier implementation.